vsix/ProjectTemplates/VC/Windows Universal/BlankApp/MainPage.cpp

```diff
@@ -7,11 +7,6 @@ using namespace Windows::UI::Xaml;
 
 namespace winrt::$safeprojectname$::implementation
 {
-    MainPage::MainPage()
-    {
-        InitializeComponent();
```

Comment:
Removing this will be confusing for developers thinking that this is missing and they will add it back. Can we keep the constructor with a comment that cppwinrt automatically calls InitializeComponent() ?

Reply:
That's why I wasn't a fan, it changes long-established practice (in both C# and C++/CX)

Reply:
That's why I wasn't a fan of the original example - making it different - but this now just removes boilerplate code and doesn't replace it with anything - it's a net win.

Reply:
Yep, that's a win, but should be documented somewhere (as well as the need to "override" InitializeComponent if XAML access during construction (for lack of a better word) is required).

Reply:
My concern with adding the ctor back is that it might encourage users who were previously relying on it to then call a property accessor, like myButton().  That would return null without a preceding call to InitializeComponent, which we don't want to encourage users to add to the ctor.

Reply:
If the ctor is blank with a comment mentionning the change and  how to override InitializeComponent, I don't think it would be an issue (but then the comment would need to be localized). Users can then delete it if desired.

Reply:
Added comments and a link to documentation, per feedback from all.  Long-term, I'll also get InitializeComponent covered on docs.microsoft.com.

@jlaanstra, Unfortunately, there's no good way to determine whether a newer VSIX item (from template) is paired with an older version of C++/WinRT nuget without false positives/negatives.  Ideally, we'd warn in both scenarios - when InitializeComponent was *not* being called from the constructor and should be, and when it is and should not be.  A developer has two remedies - update the CppWinRT nuget, or add InitializeComponent to the constructor.  We'd prefer the former, but they could do the latter. And if they didn't want to be nagged with a false-positive, we'd need to add a #define to ignore?  Tabling this for now.